### PR TITLE
[Improvement][SET-PROPERTY]Support for set query_timeout property 

### PR DESCRIPTION
**Pull request by DongLiang-0**
# Proposed changes

Issue Number: close #xxx

## Problem summary
[Improvement][SET-PROPERTY]Support for set query_timeout property 

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserProperty.java [hunk 1]
```diff
@@ -452,6 +469,9 @@ public List<List<String>> fetchProperty() {
         // exec mem limit
         result.add(Lists.newArrayList(PROP_EXEC_MEM_LIMIT, String.valueOf(commonProperties.getExecMemLimit())));
 
+        // timeout limit
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserProperty.java:472]:
```suggestion
        // query timeout
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 2]
```diff
@@ -146,6 +146,12 @@ public class ConnectContext {
 
     private SessionContext sessionContext;
 
+    private long userQueryTimeout;
+
+    public void setUserQueryTimeoutMap(long queryTimeout) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:151]:
```suggestion
    public void setUserQueryTimeout(long queryTimeout) {
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 3]
```diff
@@ -557,6 +563,11 @@ public void checkTimeout(long now) {
 
                 // Only kill
                 killFlag = true;
+            } else if (delta > userQueryTimeout * 1000) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:566]: If querytimeout is set in property, it should override the querytimeou in session variable

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserProperty.java (hunk 1 above)
Comment (DongLiang-0) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserProperty.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java (hunk 3 above)
Comment (DongLiang-0) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java]: done

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 40.1 seconds
 load time: 584 seconds
 storage size: 17154768628 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221019225024_clickbench_pr_31450.html

**Comment by DongLiang-0** [comment]
@morningman PTAL

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**